### DISCUSSION TITLE
Problems connecting to ESP8266

### DISCUSSION THREAD

**Discussion by gitout** [micropython/micropython#11207] [ESP8266]
Hi, I am having problems connecting to ESP8266.  I am running Raspbian on an AML-S9055X-CC "Le Potato" connected to ESP8266 NodeMCU CP2102 ESP-12E via the USB port.   I am able to successfully flash the chip and successfully install the latest stable firmware.  ESP8266 is present as /dev/ttyUSB0,  but when I run picocom /dev/ttyUSB0,  I get "terminal ready" and then a bunch of garbage characters that keep on scrolling.  rshell cannot connect either stuck on "trying to connect to REPL".  
I have tried adjusting baud rates, flashing with different --flash_mode values, different USB cables, but no success.  The power supply is a dedicated power supply for the board.   Other than a hardware issue, is there anything else I could try?

**Comment by slash128v6** [comment]
Hi, I am having a similar issue. Originally, I was able to flash them and they worked fine last weekend. I have several NodeMCU ESP8266 dev boards that exhibit the same symptoms. During this week they starting crashing and restarting via wdt reset. I have tried reflashing with different cables, different baud rates, different flash mode values and even 2 other laptops with fresh Python, PIP and Micropython installs. I have even tried a couple brand new dev boards out of the foil wrappers and have also tried a couple Wemos D1 Mini boards, all with same result. These boards all work fine with Arduino.

I have some ESP32 dev boards that are working fine with Micropython. It's just the ESP8266 flavors that I am having trouble with.

**Reply by gitout** [reply]
@slash128v6 I resolved the issue, at least for me. Thinking it was a hardware issue, I ordered some esp8266 boards from another manufacturer. Flashing was successful, but again, rshell was stuck on "trying to connect to REPL". I also used another computer with the same results. I then looked at this thread (where you also had responded): https://github.com/orgs/micropython/discussions/10974 and found the answer provided by @dnguyen76 worked. Specifically, the problem was in the flashing procedure. Using the following worked for me:

esptool.py --port /dev/ttyUSB0 write_flash --flash_size=detect -fm dio 0x00000 esp8266-20220618-v1.19.1.bin

The crucial key here for me is the argument "--flash_size=detect"

Omitting this argument would result in the aforementioned problems, including this argument would result in a successful flashing and able to connect to the board via rshell. I also reproduced this with the other ESP8266 board with the same successful result. Hope this helps.

**Comment by slash128v6** [comment]
Hi @gitout, thank you for messaging. I had tried that flash command as well but unfortunately it didn't help in my case. I am starting to wonder if it is something specific to the esptool software on Windows that may have changed recently? When I first started with Micropython with a NodeMCU esp8266 dev board the weekend before last I was able to flash 2 boards from a Windows 11 laptop and work with them no problem. Then at some point during the week I started getting random resets and garbage in the REPL. I tried another dev board brand new out of the foil with the same result and then tried a Wemos D1 mini, same result. I tried on 2 other Windows laptops with fresh installs of esptool, one was Windows 10 and the other Windows 11, all with the same result. All these boards work fine with Arduino, which I know doesn't necessarily mean anything other than the hardware seems fine. Then late last week I was curious about trying from Linux so I spun up an Ubuntu VM and these same boards are actually working now. When reconnecting to the Windows host (same laptop) the resets and garbage start again. Connect to the Linucx VM and they work fine...

**Reply by slash128v6** [reply]
One other note, I have some ESP32 dev boards that are working fine with my Windows laptops. It seems my issues are isolated to the ESP8266 boards.

**Reply by gitout** [reply]
I as well started out with Windows but switched to Linux due to the instability and odd issues Windows seems to have with COM ports.  Sounds like you have had success with Linux so that is good.
